### PR TITLE
Comparator: Add gpio_loopback based test suite

### PR DESCRIPTION
**Pull request by bjarki-andreasen**
Add test suite which uses GPIO loopback to produce a "very low" and "very high" voltage at the positive input of the comparator using the output of a GPIO. The negative input is set to a voltage between the high and low output voltages of the GPIO using whichever internal reference is available to the comparator.

### CODE REVIEW

File: tests/drivers/comparator/gpio_loopback/testcase.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  drivers.comparator.gpio_loopback.mcux_acmp:
+    platform_allow:
+      - frdm_ke15z
+  drivers.comparator.gpio_loopback.nrf_comp:
+    extra_args:
+      - SNIPPET="gpio_loopback_nrf_comp"
+    platform_allow:
+      - nrf54h20dk/nrf54h20/cpuapp
+      - nrf54l15dk/nrf54l15/cpuapp
+      - nrf54l15pdk/nrf54l15/cpuapp
```
Comment (nordic-piks) [inline tests/drivers/comparator/gpio_loopback/testcase.yaml:20]: nrf54l15dk is the only supported i.e. please remove nrf54l15pdk

File: tests/drivers/comparator/gpio_loopback/testcase.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+      - nrf5340dk/nrf5340/cpuapp
+  drivers.comparator.gpio_loopback.nrf_lpcomp:
+    extra_args:
+      - SNIPPET="gpio_loopback_nrf_lpcomp"
+    platform_allow:
+      - nrf54h20dk/nrf54h20/cpuapp
+      - nrf54l15dk/nrf54l15/cpuapp
+      - nrf54l15pdk/nrf54l15/cpuapp
```
Comment (nordic-piks) [inline tests/drivers/comparator/gpio_loopback/testcase.yaml:28]: pdk should be removed.

File: tests/drivers/comparator/gpio_loopback/testcase.yaml (hunk 1 above)
Comment (fabiobaltieri) [reply tests/drivers/comparator/gpio_loopback/testcase.yaml:20]: do you folks have plan to drop the board entirely?

File: tests/drivers/comparator/gpio_loopback/testcase.yaml (hunk 2 above)
Comment (bjarki-andreasen) [reply tests/drivers/comparator/gpio_loopback/testcase.yaml]: done

File: tests/drivers/comparator/gpio_loopback/testcase.yaml (hunk 1 above)
Comment (bjarki-andreasen) [reply tests/drivers/comparator/gpio_loopback/testcase.yaml]: yep, https://github.com/zephyrproject-rtos/zephyr/pull/79944

**Review by nordic-piks — APPROVED** [review]

**Review by fabiobaltieri — DISMISSED** [review]

**Comment by bjarki-andreasen** [comment]
forgot to remove the last PDK file :)

**Review by fabiobaltieri — APPROVED** [review]

**Comment by henrikbrixandersen** [comment]
Wouldn't it be better to use a DAC output?

**Comment by bjarki-andreasen** [comment]
> Wouldn't it be better to use a DAC output?

It would be cool :) but it would require fixtures with two DACs and loop back to the differential comparator inputs, which is not exactly common :) I went with this simpler GPIO loopback based one to be as portable as possible :)

**Review by henrikbrixandersen — APPROVED** [review]

**Pull request merged**